wolfSSL/src/chapter02.md

```diff
@@ -2470,3 +2470,627 @@ PORT Lists:
 * Zephyr
 * Mynewt
 * INTIME-RTOS
+
+## Building For NXP CAAM
+    
+### i.MX8 (Linux)
+    
+#### Known Issues\
+- If exiting an open HSM key store session before closing up the HSM session (wc_SECO_CloseHSM and wolfSSL_CLeanup or wolfCrypt_Cleanup) the next time the NVM is started up it segfaults. A power cycle is needed to work around it.
+
+#### Limitations Found\
+- AES operations with large inputs (i.e. 1 megabyte) fails with the SECO getting “Not enough space in shared memory”.
+- After creating 2 keystores the attempt to create a 3rd will fail. To reset the keystores do rm -rf /etc/seco_hsm and power cycle the device.
+
+#### Intro\
+On i.MX8 devices there is a SECO hardware module available for heightened security. This module handles AES operations and key storage, limited ECC operations and key storage, and provides a RNG. wolfSSL has been expanded to make use of the SECO where possible. For some algorithms the CAAM on i.MX8 supports them but the SECO module does not yet. In these cases wolfSSL will make calls through /dev/crypto down to a Linux CAAM driver that creates jobs for the CAAM directly. There are some algorithms supported by default with the NXP Linux CAAM driver but not all algorithms that the CAAM supports. wolfSSL has expanded the Linux CAAM driver to add support for additional algorithms. To use both avenues of accessing the CAAM from the same application the “devId” associated with the different code paths can be set, either WOLFSSL_CAAM_DEVID or WOLFSSL_SECO_DEVID. These IDs are used when first initializing a structure to set which code path will be used throughout the lifetime of the structure. If using software only then the default INVALID_DEVID should be set. The exception to this is SECO items that do not use the key store; TRNG, and Hashing.
+\
+
+Versions of software used:\
+
+- imx-seco-libs branch imx_5.4.24_2.1.0
+- NXP “repo” tool and Yocto build. Documentation on Yocto setup is here
+- wolfSSL 5.2.0 + (was developed after 5.2.0 release)
+
+
+#### Supported Algorithms\
+List the supported algorithms and any restrictions
```

Comment:
This line sounds like a request.  Should this be changed to a statement: "Supported algorithms, modes, and operations include:"?